src/main/java/com/commercetools/sync/commons/utils/CustomUpdateActionUtils.java

```diff
@@ -144,7 +146,18 @@
                 final String errorMessage = format(CUSTOM_FIELDS_UPDATE_ACTIONS_BUILD_FAILED,
                     resourceTypeIdGetter.apply(oldResource),
                     resourceIdGetter.apply(oldResource), exception.getMessage());
-                syncOptions.applyErrorCallback(errorMessage, exception);
+
+                if (((ParameterizedType) syncOptions.getClass().getGenericSuperclass())
+                    .getActualTypeArguments()[0].getClass().isInstance(oldResource)) {
```

Comment:
Why we need this check?

Reply:
Old/new resource can be different than declared in syncOptions. F.e: 
`ProductSyncOptions extends `BaseSyncOptions<Product, ProductDraft>` so callbacks are expecting `Product` type as oldResource and `ProductDraft` as newResource. But there are situations where they are different: `Asset`/`AssetDraft`, `ProductVariant`/`ProductVariantDraft` etc. I will add comment in source code to clarify this.

Reply:
It does not seem right that we would not pass anything in case of Assets issue besides the message. Is there a test for that? Also how could it be `ProductVariant/ProductVariantDraft` if those use product type attributes and not custom fields?

Reply:
Tests:
- CustomUpdateActionUtilsTest, ProductVariantAssetUpdateActionUtilsTest -are using `Asset`
- BuildProductVariantPricesUpdateActionsTest - is using `ProductVariant`